lib/shelley/src/Cardano/Wallet/Shelley/Api/Server.hs

```diff
@@ -487,13 +493,22 @@ server byron icarus shelley multisig spl ntp =
                 FetchSMASH smashServer -> getHealth smashServer
                 _ -> pure (ApiHealthCheck NoSmashConfigured)
 
-    sharedWallets :: Server SharedWallets
-    sharedWallets =
-             postSharedWallet multisig generateKeyFromSeed ShelleyKey
-        :<|> (fmap fst . getWallet multisig mkSharedWallet)
-        :<|> patchSharedWallet multisig ShelleyKey Payment
-        :<|> patchSharedWallet multisig ShelleyKey Delegation
-        :<|> deleteWallet multisig
+    sharedWallets
+        :: ApiLayer (SharedState n SharedKey) SharedKey
```

Comment:
The other server functions get their `ctx` from the function scope.
These ones could also, right?

Reply:
the problem is that it takes ctx type from the above context and we have ApiLayer (SharedStaten ShelleyKey) ShelleyKey ... so I needed to do this trick to make it tick...Nevertheless, I will relook if I can do something in coming PR....